Added support for coordinate systems: old GSM as Static, real GSM, SM, GSE and time dependent GSM and SM

ElonOlsson opened the pull request:
Making an initial pull request to get it rolling, though the transforms_GSM_SM_TimeDependent.asset 4 files needs to be added to server first if we're allowed. I left it as is for now.

Appreciate feedback on naming of transform assets and the meta tag in the assets. Please try it out 
![Screenshot (80)](https://github.com/user-attachments/assets/e8ddecfc-87a3-4d80-8164-bedc2d1ebeef)

ylvaselling commented on data/assets/examples/debugcoordinateaxes.asset:
```diff
@@ -1,11 +1,106 @@
+  },
+  GUI = {
+    Name = "GSM Axes",
+    Path = "/Other/Coordinate Systems"
```
```suggestion
    Path = "/Examples/Coordinate Systems"
```

ylvaselling commented on data/assets/examples/debugcoordinateaxes.asset:
```diff
@@ -1,11 +1,106 @@
+  },
+  GUI = {
+    Name = "SM-Time Axes",
+    Path = "/Other/Coordinate Systems"
```
```suggestion
    Path = "/Examples/Coordinate Systems"
```

ylvaselling commented on data/assets/examples/debugcoordinateaxes.asset:
```diff
@@ -1,11 +1,106 @@
+  },
+  GUI = {
+    Name = "GSM-Time Axes",
+    Path = "/Other/Coordinate Systems"
```
```suggestion
    Path = "/Examples/Coordinate Systems"
```

ylvaselling commented:
Tried the debugcoordinatesaxes.asset and got these errors
![image](https://github.com/user-attachments/assets/5b6951d5-5f52-4614-8da1-971335243036)

Tried the magnetosphere asset and got these errors
![image](https://github.com/user-attachments/assets/25cc2f60-2182-4e17-8ada-1da2f904fc4a)

alexanderbock commented:
Did you intend it to be in the examples folder?  I thought that we'd move them somewhere into the scene/solarsystem/planets/earth?

ElonOlsson commented:
For the transforms assets, I moved them to be under planets/earth instead of in planets/earth/magnetosphere like you suggested Alex.
I didn't move the debugcoordinateaxes asset and just added the sgn in the same manner as the ones already in there. We can absolutely move the GUI path to Examples instead of Other.
Your error with the GSM and also assuming SM with time in the name you need the kernels, which I assume we want to put on the server first. The other error, Ylva, seems like you just don't have the class built with cmake?

alexanderbock commented on data/assets/examples/debugcoordinateaxes.asset:
```diff
@@ -1,11 +1,107 @@
+  Identifier = "GSECoordinateAxes",
+  Parent = GSE_Transforms.GSEReferenceFrame.Identifier,
+  Transform = {
+    Scale ={
```
```suggestion
    Scale = {
```

alexanderbock commented on data/assets/examples/debugcoordinateaxes.asset:
```diff
@@ -1,11 +1,107 @@
+  Identifier = "GSMCoordinateAxes",
+  Parent = GSM_SM_Transforms.GSMReferenceFrame.Identifier,
+  Transform = {
+    Scale ={
```
```suggestion
    Scale = {
```

alexanderbock commented on data/assets/examples/debugcoordinateaxes.asset:
```diff
@@ -1,11 +1,106 @@
 local earthTransforms = asset.require("scene/solarsystem/planets/earth/transforms")
```
I'm not sure if we need this as an example file in this location at all tbh?  It's not really serving as an example for debugaxes, but for the coordinate system.  And there there 'main' asset file in the scene subfolder would do the same thing in my view

alexanderbock commented on data/assets/examples/debugcoordinateaxes.asset:
```diff
@@ -1,11 +1,107 @@
+  Identifier = "SMCoordinateAxes",
+  Parent = GSM_SM_Transforms.SMReferenceFrame.Identifier,
+  Transform = {
+    Scale ={
```
```suggestion
    Scale = {
```

alexanderbock commented on data/assets/examples/debugcoordinateaxes.asset:
```diff
@@ -1,11 +1,107 @@
+  Identifier = "SMTimeCoordinateAxes",
+  Parent = GSM_SM_Time_Transforms.SMTimeReferenceFrame.Identifier,
+  Transform = {
+    Scale ={
```
```suggestion
    Scale = {
```

alexanderbock commented on data/assets/examples/debugcoordinateaxes.asset:
```diff
@@ -1,11 +1,107 @@
+  Identifier = "GSMTimeCoordinateAxes",
+  Parent = GSM_SM_Time_Transforms.GSMTimeReferenceFrame.Identifier,
+  Transform = {
+    Scale ={
```
```suggestion
    Scale = {
```

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/kernels/ECLIPDATE.tf:
```diff
@@ -0,0 +1,56 @@
+Earth Mean Ecliptic and Equinox of Date frame (ECLIPDATE)
```
We'll want to move all of these files into an `identifier` based `asset.resource` as well

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/magnetosphere/magnetosphere.asset:
```diff
@@ -28,8 +28,8 @@ local EarthMagnetosphere = {
     ColorTableRanges = { { 50, 300 } },
-    ColorTablePaths = { transferFunction },
     LoadAtRuntime = true,
+    ColorTablePaths = {transferFunction},
```
```suggestion
    ColorTablePaths = { transferFunction },
```

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/transforms_GSE.asset:
```diff
@@ -0,0 +1,51 @@
+local transforms = asset.require("scene/solarsystem/planets/earth/transforms")
```
The filename should be `transforms_gse.asset` as we only do lowercase filenames

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/transforms_GSE.asset:
```diff
@@ -0,0 +1,51 @@
+local coreKernels = asset.require("spice/core")
+local Eclipdate = asset.resource("kernels/ECLIPDATE.tf")
+local GSEKernel = asset.resource("kernels/GSE.tf")
+
```
```suggestion


```

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/transforms_GSE.asset:
```diff
@@ -0,0 +1,51 @@
+local Frame = {
+  GSE = "GSE"
+}
+local GSEReferenceFrame = {
```
```suggestion

local GeocentricSolarEcliptic = {
```

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/transforms_GSE.asset:
```diff
@@ -0,0 +1,51 @@
+  GSE = "GSE"
+}
+local GSEReferenceFrame = {
+  Identifier = "GSEReferenceFrame",
```
```suggestion
  Identifier = "GeocentricSolarEcliptic",
```

No need to say ReferenceFrame as it doesn't make a different to a user whether it is just a reference frame or if it has a renderable attached to it

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/transforms_GSM_SM.asset:
```diff
@@ -0,0 +1,74 @@
+local transforms = asset.require("scene/solarsystem/planets/earth/transforms")
```
Filename -> `transforms_gsm_sm.asset`

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/transforms_GSM_SM.asset:
```diff
@@ -0,0 +1,74 @@
+local transforms = asset.require("scene/solarsystem/planets/earth/transforms")
+local coreKernels = asset.require("spice/core")
+
```
```suggestion



```

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/transforms_GSM_SM.asset:
```diff
@@ -0,0 +1,74 @@
+local GSM_Kernel = asset.resource("kernels/GSM.tf")
+local SM_Kernel = asset.resource("kernels/SM.tf")
+
+local Frame = {
```
```suggestion

local Frame = {
```

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/transforms_GSM_SM.asset:
```diff
@@ -0,0 +1,74 @@
+}
+
+local GSMReferenceFrame = {
+  Identifier = "GSMReferenceFrame",
```
```suggestion
  Identifier = "GeocentricSolarMagnetospheric",
```

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/transforms_GSM_static.asset:
```diff
@@ -0,0 +1,52 @@
+    }
+  },
+  GUI = {
+    Name = "Static GSM Reference Frame",
```
```suggestion
    Name = "Geocentric Solar Magnetospheric Static",
```

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/transforms_GSM_static.asset:
```diff
@@ -0,0 +1,52 @@
+  }
+}
+
+
```
```suggestion
```

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/transforms_GSM_static.asset:
```diff
@@ -0,0 +1,52 @@
+
+asset.onInitialize(function()
+  openspace.spice.loadKernel(GSM_Static_Kernel)
+  openspace.addSceneGraphNode(GSMStaticReferenceFrame)
```
```suggestion
  openspace.addSceneGraphNode(GeocentricSolarMagnetosphericStatic)
```

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/transforms_GSM_static.asset:
```diff
@@ -0,0 +1,52 @@
+end)
+
+asset.onDeinitialize(function()
+  openspace.removeSceneGraphNode(GSMStaticReferenceFrame)
```
```suggestion
  openspace.removeSceneGraphNode(GeocentricSolarMagnetosphericStatic)
```

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/transforms_GSM_static.asset:
```diff
@@ -0,0 +1,52 @@
+end)
+
+asset.export("Frame", Frame)
+asset.export(GSMStaticReferenceFrame)
```
```suggestion
asset.export(GeocentricSolarMagnetosphericStatic)
```

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/transforms_GSM_static.asset:
```diff
@@ -0,0 +1,52 @@
+
+
+asset.meta = {
+Name = "Earth static GSM transforms",
+Version = "1.0",
+Description = "A static version of Earth transforms: GSMReferenceFrame. Geocentric Solar Magnetospheric",
+Author = "CCMC",
+URL = "http://openspaceproject.com",
+License = "MIT license"
```
```suggestion
  Name = "Earth static GSM transforms",
  Version = "1.0",
  Description = "A static version of Earth transforms: GSMReferenceFrame. Geocentric Solar Magnetospheric",
  Author = "CCMC",
  URL = "http://openspaceproject.com",
  License = "MIT license"
```

alexanderbock commented on data/assets/scene/solarsystem/sun/transforms_heliosphere.asset:
```diff
@@ -27,26 +28,71 @@ local HEEQ180ReferenceFrame = {
   }
 }
 
+local HEEQReferenceFrame = {
```
```suggestion
local HeliocentricEarthEquatorial = {
```

alexanderbock commented on data/assets/scene/solarsystem/sun/transforms_heliosphere.asset:
```diff
@@ -27,26 +28,71 @@ local HEEQ180ReferenceFrame = {
 }
 
+local HEEQReferenceFrame = {
+  Identifier = "HEEQReferenceFrame",
```
```suggestion
  Identifier = "HeliocentricEarthEquatorial",
```

alexanderbock commented on data/assets/scene/solarsystem/sun/transforms_heliosphere.asset:
```diff
@@ -27,26 +28,71 @@ local HEEQ180ReferenceFrame = {
+    }
+  },
+  GUI = {
+    Name = "HEEQ Reference Frame",
```
```suggestion
    Name = "Heliocentric Earth Equatorial",
```

alexanderbock commented on data/assets/scene/solarsystem/sun/transforms_heliosphere.asset:
```diff
@@ -27,26 +28,71 @@ local HEEQ180ReferenceFrame = {
+  }
+}
+
+local HEEReferenceFrame = {
```
```suggestion
local HeliocentricEarthEcliptic = {
```

alexanderbock commented on data/assets/scene/solarsystem/sun/transforms_heliosphere.asset:
```diff
@@ -27,26 +28,71 @@ local HEEQ180ReferenceFrame = {
+}
+
+local HEEReferenceFrame = {
+  Identifier = "HEEReferenceFrame",
```
```suggestion
  Identifier = "HeliocentricEarthEcliptic",
```

alexanderbock commented on data/assets/scene/solarsystem/sun/transforms_heliosphere.asset:
```diff
@@ -27,26 +28,71 @@ local HEEQ180ReferenceFrame = {
+    }
+  },
+  GUI = {
+    Name = "HEE Reference Frame",
```
```suggestion
    Name = "Heliocentric Earth Ecliptic",
```

alexanderbock commented on data/assets/scene/solarsystem/sun/transforms_heliosphere.asset:
```diff
@@ -27,26 +28,71 @@ local HEEQ180ReferenceFrame = {
 
   openspace.addSceneGraphNode(HEEQ180ReferenceFrame)
+  openspace.addSceneGraphNode(HEEQReferenceFrame)
+  openspace.addSceneGraphNode(HEEReferenceFrame)
```
```suggestion
  openspace.addSceneGraphNode(HeliocentricEarthEcliptic)
```

alexanderbock commented on data/assets/scene/solarsystem/sun/transforms_heliosphere.asset:
```diff
@@ -27,26 +28,71 @@ local HEEQ180ReferenceFrame = {
 end)
 
 asset.onDeinitialize(function()
+  openspace.removeSceneGraphNode(HEEReferenceFrame)
```
```suggestion
  openspace.removeSceneGraphNode(HeliocentricEarthEcliptic)
```

alexanderbock commented on data/assets/scene/solarsystem/sun/transforms_heliosphere.asset:
```diff
@@ -27,26 +28,71 @@ local HEEQ180ReferenceFrame = {
 
 asset.onDeinitialize(function()
+  openspace.removeSceneGraphNode(HEEReferenceFrame)
+  openspace.removeSceneGraphNode(HEEQReferenceFrame)
```
```suggestion
  openspace.removeSceneGraphNode(HeliocentricEarthEquatorial)
```

alexanderbock commented on data/assets/scene/solarsystem/sun/transforms_heliosphere.asset:
```diff
@@ -27,26 +28,71 @@ local HEEQ180ReferenceFrame = {
+  openspace.spice.loadKernel(heeKernel)
 
   openspace.addSceneGraphNode(HEEQ180ReferenceFrame)
+  openspace.addSceneGraphNode(HEEQReferenceFrame)
```
```suggestion
  openspace.addSceneGraphNode(HeliocentricEarthEquatorial)
```

alexanderbock commented on data/assets/scene/solarsystem/sun/transforms_heliosphere.asset:
```diff
@@ -1,10 +1,11 @@
 
-local kernel = asset.resource("kernels/HEEQ180.tf")
-
 local HEEQ180ReferenceFrame = {
```
```suggestion
local HeliocentricEarthEquatorial180 = {
```

alexanderbock commented on data/assets/scene/solarsystem/sun/transforms_heliosphere.asset:
```diff
@@ -1,10 +1,11 @@
-local kernel = asset.resource("kernels/HEEQ180.tf")
-
 local HEEQ180ReferenceFrame = {
   Identifier = "HEEQ180ReferenceFrame",
```
```suggestion
  Identifier = "HeliocentricEarthEquatorial180",
```

alexanderbock replied:
Can't mark it in GitHub but the GUI.Name should be "Heliocentric Earth Equatorial"

alexanderbock commented on data/assets/scene/solarsystem/sun/transforms_heliosphere.asset:
```diff
@@ -27,26 +28,71 @@ local HEEQ180ReferenceFrame = {
 asset.onDeinitialize(function()
+  openspace.removeSceneGraphNode(HEEReferenceFrame)
+  openspace.removeSceneGraphNode(HEEQReferenceFrame)
   openspace.removeSceneGraphNode(HEEQ180ReferenceFrame)
```
```suggestion
  openspace.removeSceneGraphNode(HeliocentricEarthEquatorial180)
```

alexanderbock replied:
```suggestion
  openspace.removeSceneGraphNode(HeliocentricEarthEquatorial180)
```

alexanderbock reviewed (commented):
All of the new coordinate systems desparately needs a GUI.Description that describes what they do to a semi-lay audience. You can probably take the beginning of the frame kernel definitions and put that into the description.

Also all of the frame kernels should be uploaded to the server and then used with an `asset.resource` instead

alexanderbock reviewed (commented):
All of the new coordinate systems desparately needs a GUI.Description that describes what they do to a semi-lay audience. You can probably take the beginning of the frame kernel definitions and put that into the description.

Also all of the frame kernels should be uploaded to the server and then used with an `asset.resource` instead

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/transforms_gse.asset:
```diff
@@ -1,10 +1,15 @@
 
 
+local earthcentricKernels = asset.resource({
+  Name = "Earthcenteric frame kernels",
```
```suggestion
  Name = "Earthcentric Frame Kernels",
```

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/transforms_gsm_sm.asset:
```diff
@@ -1,10 +1,14 @@
 
 
+local earthcentricKernels = asset.resource({
+  Name = "Earthcenteric frame kernels",
```
```suggestion
  Name = "Earthcentric Frame Kernels",
```

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/transforms_gsm_sm_timedependent.asset:
```diff
@@ -4,8 +4,6 @@ local coreKernels = asset.require("spice/core")
-
 local kernels = {
   "C:/Users/eolsson/Downloads/earthnpole_19500101_20251231_v01.bsp",
   "C:/Users/eolsson/Downloads/earthnpole_runout.bsp",
```
These entries (and other like it should go as well)

alexanderbock commented on data/assets/scene/solarsystem/planets/earth/transforms_gsm_static.asset:
```diff
@@ -1,9 +1,14 @@
 
 
+local earthcentricKernels = asset.resource({
+  Name = "Earthcenteric frame kernels",
```
```suggestion
  Name = "Earthcentric Frame Kernels",
```

alexanderbock commented:
@ElonOlsson
